### PR TITLE
docs: Add landing page for React Native Screens

### PR DESCRIPTION
**Pull request by patrycjakalinska**
This PR adds landing page for React Native Screens with the use of @swmansion/t-rex-ui component library.

The mobile view right now is done without proper design - let me know have some ideas on it!

It's open for testing but keep in mind mentioned to-be-fixed things.

- [x] Add favicon
- [x] Add og-image
- [x] Add hero animation
- [x] Add Multiplatform tile animation
- [ ] (optional) Waiting for proper mobile designs
- [x] Bump @swmansion/t-rex-ui to latest

Laptop:

https://github.com/software-mansion/react-native-screens/assets/59940332/486b82cd-e2c6-4e33-b72e-cfc3de44a08c

Mobile:

https://github.com/software-mansion/react-native-screens/assets/59940332/af8cd946-b814-4071-97a8-8650de139312

Animacja na hero:

https://github.com/user-attachments/assets/fe506eb7-098d-4c70-b403-1a70e4374f2b

### CODE REVIEW

File: docs/static/img/favicon.ico
Comment (tboba) [inline docs/static/img/favicon.ico:1]: Personally, I think we could enhance the favicon, so that it could have better appearance on browsers with dark mode.

I have two suggestions there:
- Let's make the background transparent (it should look decent with the foreground "screen" in the logo, but dunno how the background "screen" will look,
- Let's make the corners of background rounded, same as in reanimated documentation.

image

File: docs/README.md
Comment (tboba) [inline docs/README.md:1]: Do we wanna keep original README, or modify it, so that it'll contain information about the landing page, similar as [here](https://github.com/software-mansion/react-native-reanimated/blob/main/packages/docs-reanimated/README.md)?

File: docs/docs/intro.md
Comment (tboba) [inline docs/docs/intro.md:1]: Is this file required? 😄

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// Note: type annotations allow type checking and IDEs autocompletion
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: 'React Native Screens',
+  favicon: 'img/favicon.ico',
+
+  url: 'https://docs.swmansion.com',
+
+  baseUrl: '/',
```
Comment (tboba) [inline docs/docusaurus.config.js:11]: Probably?

```suggestion
  baseUrl: '/react-native-screens/',
```

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  organizationName: 'software-mansion',
+  projectName: 'react-native-screens',
+
+  onBrokenLinks: 'throw',
+  onBrokenMarkdownLinks: 'warn',
```
Comment (tboba) [inline docs/docusaurus.config.js:17]: I'm wondering if we should `throw` an error, instead of warning if Docusaurus detects broken md links?

File: docs/sidebars.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      type: 'category',
+      label: 'Tutorial',
+      items: ['tutorial-basics/create-a-document'],
+    },
+  ],
+   */
+};
+
+module.exports = sidebars;
```
Comment (tboba) [inline docs/sidebars.js:33]: Dunno if we're using this component somewhere?

File: docs/src/components/FooterBackground/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+@media (max-width: 375px) {
+  .container {
+    margin-top: 173px;
+  }
+  footer {
+    margin-top: -173px;
+  }
+}
```
Comment (tboba) [inline docs/src/components/FooterBackground/styles.module.css:57]: I'm concerned why these values are so deviated 😄 Can we round them?

File: docs/src/components/Hero/ScreensIcon/Icon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+              : (hoveredIndex === index - 1 || hoveredIndex === index + 1) &&
+                  styles.neighbourHovered
+          )}
+          onMouseEnter={() => setHoveredIndex(index)}
+          onMouseLeave={() => setHoveredIndex(null)}>
+          {screen.map((path, i) => (
+            <path
+              key={i}
+              fill={
+                hoveredIndex === index && windowWidth > 996
```
Comment (tboba) [inline docs/src/components/Hero/ScreensIcon/Icon/index.tsx:131]: Why are we blocking "filling" effect on hover? Are there any side-effects on smaller displays? 🤔

File: docs/src/components/Hero/StartScreen/index.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            <span>Screens</span>
+          </h1>
+          <h2 className={styles.subheadingLabel}>
+            Native Navigation primitives for your React Native app.
+          </h2>
+        </div>
+        <div className={styles.buttonContainer}>
+          <HomepageButton
+            target="_blank"
+            href="https://reactnavigation.org/docs/screen"
```
Comment (tboba) [inline docs/src/components/Hero/StartScreen/index.tsx:22]: I'd go for redirecting [here](https://reactnavigation.org/docs/7.x/native-stack-navigator) (or at least redirecting to 7.x version of that article) but I'm leaving a space for discussion there.

File: docs/src/components/Overview/Multiplatform/styles.module.css [hunk 7]
```diff
@@ -0,0 +1,53 @@
+.multiplatformContainer {
+  background-color: var(--swm-navy-dark-40);
+  border-radius: 16px;
+  width: 456px;
+  height: 345px;
```
Comment (tboba) [inline docs/src/components/Overview/Multiplatform/styles.module.css:5]: I'd do `450px` and `350px` 🤔

File: docs/src/components/Overview/OverviewTile/styles.module.css [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  gap: 1.5rem;
+  flex: 1;
+}
+
+.contentContainer h3 {
+  font-size: 32px;
+  line-height: 36.8px;
```
Comment (tboba) [inline docs/src/components/Overview/OverviewTile/styles.module.css:21]: Too deviated value for line-height 😄 Let's leave it as `36px`.

File: docs/src/components/Hero/ScreensIcon/Icon/index.tsx [hunk 9]
```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import useScreenSize from '@site/src/hooks/useScreenSize';
+import styles from './styles.module.css';
+
+const screens = [
```
Comment (tboba) [inline docs/src/components/Hero/ScreensIcon/Icon/index.tsx:6]:
```suggestion
const SCREENS = [
```

File: docs/src/components/Hero/ScreensIcon/Icon/index.tsx [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+          ? 0.4
+          : windowWidth > 2000
+          ? 0.2
+          : 0.3)
+      }
+      fill="none"
+      viewBox="0 0 468 397"
+      xmlns="http://www.w3.org/2000/svg"
+      className={styles.svg}>
+      {screens.map((screen, index) => (
```
Comment (tboba) [inline docs/src/components/Hero/ScreensIcon/Icon/index.tsx:115]:
```suggestion
      {SCREENS.map((screen, index) => (
```

File: docs/src/components/Hero/ScreensIcon/Icon/index.tsx
Comment (tboba) [inline docs/src/components/Hero/ScreensIcon/Icon/index.tsx:1]: Hmm, `ScreensSequence` would be a better name for that component, I think (and `Icon` as `Sequence`)

File: docs/src/components/Overview/styles.module.css [hunk 11]
```diff
@@ -0,0 +1,20 @@
+.overviewContainer {
+  display: flex;
+  flex-direction: column;
+  gap: 6.5rem;
+}
+
+.title {
+  font-size: 48px;
+  font-weight: 700;
+  line-height: 55.2px;
```
Comment (tboba) [inline docs/src/components/Overview/styles.module.css:10]:
```suggestion
  line-height: 56px;
```

File: docs/src/components/Testimonials/TestimonialItem/index.tsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  src: string;
+}
+interface Props extends PropsWithChildren {
+  author: string;
+  company?: string;
+  link: string;
+  image: ImageProps;
+}
+
+const TestimonialItem = ({ author, company, image, link, children }: Props) => {
```
Comment (tboba) [inline docs/src/components/Testimonials/TestimonialItem/index.tsx:18]: Where are we putting these Testimonials?

File: docs/src/components/Testimonials/TestimonialItem/styles.module.css [hunk 13]
```diff
@@ -0,0 +1,110 @@
+.testimonialItem {
+  background-color: var(--swm-off-white);
+  box-shadow: -12px 12px 0px var(--swm-navy-light-60);
+  padding: 1.5em;
+  position: relative;
+  width: 100%;
+  display: flex;
+  gap: 1rem;
+  flex-direction: column;
+  max-width: calc(50% - 3.5rem / 2);
```
Comment (tboba) [inline docs/src/components/Testimonials/TestimonialItem/styles.module.css:10]: Can we put a comment what we're calculating there?

File: docs/src/components/Testimonials/TestimonialList/index.tsx [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+            company={items[i + 1].company}
+            image={items[i + 1].image}
+            link={items[i + 1].link}
+            author={items[i + 1].author}>
+            {items[i + 1].body}
+          </TestimonialItem>
+        )}
+      </div>
+    );
+  }
```
Comment (tboba) [inline docs/src/components/Testimonials/TestimonialList/index.tsx:97]: I think this component is quite too JS'ish 😄 I'd refactor that (remove adding event listeners, also I have no idea why we're looping here every two items?)

File: docs/src/components/Testimonials/TestimonialList/styles.module.css [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+  width: 100%;
+  margin-top: 2.5rem;
+  justify-content: center;
+}
+
+.dot {
+  cursor: pointer;
+  height: 12px;
+  width: 12px;
+  background-color: #33488e40;
```
Comment (tboba) [inline docs/src/components/Testimonials/TestimonialList/styles.module.css:38]: Let's put that color to colors.css file (or change it to the color from our palette?)

File: docs/static/.nojekyll
Comment (tboba) [inline docs/static/.nojekyll:1]: Dunno if this file is required? 🤔

**Review by tboba — CHANGES_REQUESTED** [review]
Good job 🥳 ❤️ Just left a bunch of comments there.

File: docs/src/components/Testimonials/TestimonialList/index.tsx (hunk 14 above)
Comment (patrycjakalinska) [reply docs/src/components/Testimonials/TestimonialList/index.tsx:97]: Because two Items are always shown, and on resize they adjust to screen, it's the same as in other documentations.  It is necessary for smooth, well calculated experience with Testimonials (not all are the same height, depending on how long is the text)

File: docs/src/components/Testimonials/TestimonialList/styles.module.css (hunk 15 above)
Comment (patrycjakalinska) [reply docs/src/components/Testimonials/TestimonialList/styles.module.css:38]: I'll move it to colors.css then, because we don't have transparency colors in basic colors.css

File: docs/sidebars.js (hunk 3 above)
Comment (patrycjakalinska) [reply docs/sidebars.js:33]: It is required by docusaurus

File: docs/src/components/FooterBackground/styles.module.css (hunk 4 above)
Comment (patrycjakalinska) [reply docs/src/components/FooterBackground/styles.module.css:57]: They fix the gradient on the footer, so they are so specific to be perfect. Before it was connected to Footer component from Docusaurus so there was no need to do that, right now we need to use negative margin to move gradient down to cover footer on each media screen size

File: docs/src/components/Overview/OverviewTile/styles.module.css (hunk 8 above)
Comment (patrycjakalinska) [reply docs/src/components/Overview/OverviewTile/styles.module.css:21]: It was taken from the designs, it is common to use it among our documentations, but i'll change it if'd like

File: docs/src/components/Testimonials/TestimonialItem/index.tsx (hunk 12 above)
Comment (patrycjakalinska) [reply docs/src/components/Testimonials/TestimonialItem/index.tsx:18]: Like, where on the page? Above sponsors, but they will be there once we gather the testimonials in the future i suppose?

Comment (tboba) [reply docs/src/components/Testimonials/TestimonialItem/index.tsx:18]: Sure, I was just asking if we didn't forget about them 😄

File: docs/src/components/Overview/OverviewTile/styles.module.css (hunk 8 above)
Comment (tboba) [reply docs/src/components/Overview/OverviewTile/styles.module.css:21]: I think it shouldn't make a difference, I'm also of the opinion that final version does not have to be the same as designs 😄 `36px` would look better in code, at least imo

File: docs/docs/intro.md
Comment (patrycjakalinska) [reply docs/docs/intro.md:1]: docusaurus requires at least one file in `/docs` folder to work

File: docs/src/components/Hero/ScreensIcon/Icon/index.tsx (hunk 5 above)
Comment (patrycjakalinska) [reply docs/src/components/Hero/ScreensIcon/Icon/index.tsx:131]: we don't need any hover effects on mobile, so less than 996px it is disabled

File: docs/src/components/Overview/OverviewTile/styles.module.css [hunk 16]
```diff
@@ -0,0 +1,44 @@
+.overviewTile {
```
Comment (alduzy) [inline docs/src/components/Overview/OverviewTile/styles.module.css:1]: A grid layout may be a better solution, the gaps are inconsistent now when reversed and non-reversed items are present:

Screenshot 2024-07-30 at 09 57 15

**Review by alduzy — COMMENTED** [review]
Nice work! Left a few comments

File: docs/README.md
Comment (patrycjakalinska) [reply docs/README.md:1]: I've made some example README, you can check now 🚀

File: docs/README.md [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+
+- _Reading and sharing_ the docs with friends and co-workers
+- _Sending feedback_ in the discussion or issues on GitHub
+- _Submitting PRs_ with fixes and improvements to the docs content and styles
+
+All feedback in all forms ranging from general to tiny fixes is highly appreciated!
+
+## Meet the docs team
+
+- [Balbina Falba](https://dribbble.com/balbina): design
```
Comment (tboba) [inline docs/README.md:18]:
```suggestion
- [Balbina Fabia](https://dribbble.com/balbina): design
```

File: docs/src/components/Testimonials/TestimonialItem/styles.module.css [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+  /* max width of tile minus gap of parent */
```
Comment (tboba) [inline docs/src/components/Testimonials/TestimonialItem/styles.module.css:11]: Let's move this above `max-width` line 😄

**Review by tboba — CHANGES_REQUESTED** [review]

File: docs/src/components/FooterBackground/styles.module.css (hunk 4 above)
Comment (tboba) [reply docs/src/components/FooterBackground/styles.module.css:57]: I'm still against putting that much specific numbers, couple of pixels shouldn't break anything 😄 Let's try rounding it

File: docs/src/components/Overview/Multiplatform/styles.module.css [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+  align-items: center;
+}
+
+@keyframes marquee {
+  0% {
+    transform: translateX(0);
+  }
+  100% {
+    transform: translateX(calc(-100% - 3rem));
+    /* we need to do full full translate minus gap */
```
Comment (tboba) [inline docs/src/components/Overview/Multiplatform/styles.module.css:47]: Let's put this comment above transform line

**Review by tboba — CHANGES_REQUESTED** [review]

**Review by tboba — APPROVED** [review]

**Comment by tboba** [comment]
:shipit: :shipit: :shipit: :shipit:
